cluster-autoscaler/utils/deletetaint/delete.go

```diff
@@ -89,7 +97,20 @@ func addTaint(node *apiv1.Node, client kube_client.Interface, taintKey string, e
 			}
 			return nil
 		}
-		_, err = client.CoreV1().Nodes().Update(context.TODO(), freshNode, metav1.UpdateOptions{})
+
+		// update all taints, so we do not have to deal with adding to possibly nil taints or taint already existing
+		// patching to avoid overwriting changes that other cluster members might have done in the meantime
+		patch := []patchStruct{
+			{Op: "replace", Path: "/spec/tolerations", Value: freshNode.Spec.Taints},
+			{Op: "replace", Path: "/spec/unschedulable", Value: freshNode.Spec.Unschedulable},
+		}
+		patchJson, err := json.Marshal(patch)
+		if err != nil {
+			klog.Warningf("Error while adding %v taint on node %v: %v", getKeyShortName(taintKey), node.Name, err)
+			return err
+		}
+
+		_, err = client.CoreV1().Nodes().Patch(context.TODO(), freshNode.Name, types.JSONPatchType, patchJson, metav1.PatchOptions{})
 		if err != nil && errors.IsConflict(err) && time.Now().Before(retryDeadline) {
```

Comment:
I think a conflict can not happen when patching so this code might need a rework too